npc_gzip/utils.py

```diff
@@ -0,0 +1,87 @@
+import random
+import string
+
+from npc_gzip.exceptions import InvalidObjectTypeException
+
+
+def generate_sentence(number_of_words: int = 10) -> str:
+    """
+    Generates a sentence of random
+    numbers and letters, with 
+    `number_of_words` words in the 
+    sentence such that len(out.split()) \
+    == `number_of_words`.
+
+    Arguments:
+        number_of_words (int): The number of words you
+                               want in the sentence.
+
+    Returns:
+        str: Sentence of random numbers and letters.
+    """
+
+    if not isinstance(number_of_words, int):
+        if isinstance(number_of_words, float):
+            number_of_words = int(number_of_words)
+        else:
+            raise InvalidObjectTypeException(
+                type(number_of_words),
+                supported_types=[int, float],
+                function_name="generate_sentence",
+            )
+
+    assert number_of_words > 0, f"`number_of_words` must be greater than zero."
+
+    words = []
+    for word in range(number_of_words):
+        # initializing size of string
+        N = random.randint(1, 50)
+
+        words.append(
+            "".join(random.choices(string.ascii_uppercase + string.digits, k=N))
+        )
+
+    out = " ".join(words)
+    return out
+
+
+def generate_dataset(number_of_sentences: int) -> list:
```

Comment:
Not sure what's the purpose of this function. If it's used to replace the original `pick_samples_from_each_class_given_dataset` function, I think it misses the step of drawing samples _per class_?

Reply:
It's really just a couple of helper functions to generate data in the tests

Reply:
I see. Yeah, we can keep that and I will also add the "pick samples per class" function for convenience.